### DISCUSSION TITLE
always ask switch network to mainnet

### DISCUSSION THREAD

**Discussion by shixin-guo** [family/connectkit#10] [Q&A] [answered]
<img width="365" alt="image" src="https://user-images.githubusercontent.com/12186221/192903614-da2cd6e3-b0c6-4822-bcb0-d299dc8b7d9b.png">

When I connect with MetaMask after upgrading to 0.0.2, connectKit asks me to switch to Mainnet, but I want to use Rinkeby TestNet. Do you know what's causing this?

Because 0.0.1 is working properly, I am perplexed.😢 

this is my test code
```
import { ConnectKitProvider } from 'connectkit'
import {
  WagmiConfig,
  createClient,
  allChains,
  configureChains,
  chain,
} from 'wagmi'

import { publicProvider } from 'wagmi/providers/public'

import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
import { MetaMaskConnector } from 'wagmi/connectors/metaMask'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'

import { firebaseApp } from '@framework/provider'

const { chains, provider, webSocketProvider } = configureChains(allChains, [
  publicProvider(),
])
const client = createClient({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector(),
    new InjectedConnector({ chains: [chain.hardhat] }),
    new CoinbaseWalletConnector({
      chains,
      options: {
        appName: 'wagmi',
      },
    }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
  ],
  provider,
  webSocketProvider,
})
```

**Comment by lochie** [comment]
Hey @shixin-guo, thanks for choosing ConnectKit!

It looks like this issue has to do with using `allChains` inside your `configureChains` setup. You would need to replace `allChains` with `[chain.hardhat]` (or your list of preferred chains), and also make sure to use configured `chains` variable inside the `InjectedConnector` and `MetaMaskConnector`

Your above test code should look something like this after these changes:

```jsx
import { ConnectKitProvider } from 'connectkit'
import {
  WagmiConfig,
  createClient,
  configureChains,
  chain,
} from 'wagmi'

import { publicProvider } from 'wagmi/providers/public'

import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
import { MetaMaskConnector } from 'wagmi/connectors/metaMask'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'

import { firebaseApp } from '@framework/provider'

const appChains = [chain.hardhat];

const { chains, provider, webSocketProvider } = configureChains(appChains, [
  publicProvider(),
])
const client = createClient({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({ chains }),
    new InjectedConnector({ chains }),
    new CoinbaseWalletConnector({
      chains,
      options: {
        appName: 'wagmi',
      },
    }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
  ],
  provider,
  webSocketProvider,
})
```

I would also highly recommend, instead of the above, insist on using our [default client](https://docs.family.co/connectkit/getting-started#getting-started-implementation) implementation that handles all the above (and more) for you.

That would look something like this:

```jsx
import { WagmiConfig, createClient, chain } from "wagmi";
import { ConnectKitProvider, getDefaultClient } from "connectkit";

const alchemyId = process.env.ALCHEMY_ID;

const client = createClient(
  getDefaultClient({
    appName: "Your App Name",
    alchemyId,
    chains: [chain.hardhat]
  }),
);

const App = () => {
  return (
    <WagmiConfig client={client}>
      <ConnectKitProvider>
        /* Your App */
      </ConnectKitProvider>
    </WagmiConfig>
  );
};
```

Hoping this helps! ☺️

**Comment by shixin-guo** [comment]
@lochie 

thank you for your prompt response, Yes!!! It was a wise decision for me to use Connectkit, and I appreciate your assistance; however, it is still not working; I attempted to reproduce it in our example demo,and encountered the same problem.


code diff: in examples/cra/src/index.tsx 
<img width="1109" alt="image" src="https://user-images.githubusercontent.com/12186221/193384272-167da890-31db-4df7-84b6-2c58ac9f9088.png">

<img width="1420" alt="image" src="https://user-images.githubusercontent.com/12186221/193384258-a5200af7-311e-444b-b84c-ad99c219f86c.png">

**Comment by lochie** [comment, marked as answer]
Hey @shixin-guo,

Just letting you know that I've [talked a little about this](https://github.com/family/connectkit/issues/16#issuecomment-1264779017) over at issue #16, but your issue may also be due to using Rinkeby, since [Rinkeby is being deprecated](https://blog.ethereum.org/2022/06/21/testnet-deprecation) it's encouraged to use Goerli or Sepolia to test Ethereum in a post-merge context (you can learn more about The Merge [here](https://family.co/blog/the-merge)).

If you wish for this to be raised further I suggest [creating an issue from this discussion](https://github.com/family/connectkit/issues/new?created_from_discussion_number=10) as this is starting to become more of a debugging issue rather than a discussion about ConnectKit itself.

Thanks!

**Comment by JavLG** [comment]
Hey hey buddies,
I'm experiencing the same issues that @shixin-guo experienced recently. I'm trying to set my ConnectKit to be coupled to Goerli on dev and Mainnet on prod envs. 


I do set 2 .env:

.env.local -> Holds ALCHEMY_ID for Goerli.
.env.production -> Holds ALCHEMY_ID for Mainnet.


```ts
const appChain = env.NODE_ENV === 'production' ? chain.mainnet : chain.goerli
const alchemyId = env.NEXT_PUBLIC_ALCHEMY_ID;

    const client = createClient(
      getDefaultClient({
        appName: env.NEXT_PUBLIC_ALCHEMY_APP_NAME,
        alchemyId,
        chains: [appChain]
      }),
    );

```
Render return looks like this (Nothing uncommon):

<img width="502" alt="image" src="https://user-images.githubusercontent.com/33915099/201220281-ff6179be-178b-4da2-85c6-eae4be6dc315.png">


I still get Metamask asking to change network to mainnet on all usecases. Is this an actual bug or am I missing the documentation?

![image](https://user-images.githubusercontent.com/33915099/201220959-aaf72cb6-87a6-45f3-a470-fd656b7024f3.png)

Thanks!

**Reply by lochie** [reply]
Hey @JavLG, are you on the latest version of ConnectKit?

**Reply by JavLG** [reply]
Hey Lochie, it got fixed overnight smh. I believe it might have to do with the appName? Thats the only bit I changed so far now that I split the creds into 2 env variables. 
Versions used: 
```
"wagmi": "^0.6.7",
"connectkit": "^1.0.0",
```

Thanks for the quick response btw. Keep the good work!

**Reply by lochie** [reply]
No worries, happy to help! Feel free open an issue if you find this happening again.
Thanks and I hope you're enjoying using ConnectKit 🤝
